zipline/data/us_equity_minutes.py

```diff
@@ -11,27 +11,401 @@
+
+    def sidpath(self, sid):
+        """
+        Parameters:
+        -----------
+        sid : int
+            Asset identifier.
+
+        Returns:
+        --------
+        out : string
+            Full path to the bcolz rootdir for the given sid.
+        """
+        sid_subdir = _sid_subdir_path(sid)
+        return join(self._rootdir, sid_subdir)
+
+    def last_date_in_output_for_sid(self, sid):
+        """
+        Parameters:
+        -----------
+        sid : int
+            Asset identifier.
+
+        Returns:
+        --------
+        out : pd.Timestamp
+            The midnight of the last date written in to the output for the
+            given sid.
+        """
+        try:
+            sizes_path = "{0}/close/meta/sizes".format(self.sidpath(sid))
+            with open(sizes_path, mode='r') as f:
+                sizes = f.read()
+            data = json.loads(sizes)
+            num_days = data['shape'][0] / self._minutes_per_day
+            if num_days == 0:
+                # empty container
+                return None
+            return self._trading_days[num_days - 1]
+        except IOError:
+            return None
+
+    def _init_ctable(self, path):
+        """
+        Create empty ctable for given path.
+
+        Parameters:
+        -----------
+
+        path : string
+            The path to rootdir of the new ctable.
+        """
+        # Only create the subdir on container creation.
+        sid_dirname = os.path.dirname(path)
+        if not os.path.exists(sid_dirname):
+            os.makedirs(sid_dirname)
+        initial_array = np.empty(0, np.uint32)
+        table = ctable(
+            rootdir=path,
+            columns=[
+                initial_array,
+                initial_array,
+                initial_array,
+                initial_array,
+                initial_array,
+            ],
+            names=[
+                'open',
+                'high',
+                'low',
+                'close',
+                'volume'
+            ],
+            expectedlen=self._expectedlen,
+            mode='w',
+        )
+        table.flush()
+
+    def _zerofill(self, table, numdays):
+        num_to_prepend = numdays * self._minutes_per_day
+        prepend_array = np.zeros(num_to_prepend, np.uint32)
+        # Fill all OHLCV with zeros.
+        table.append([prepend_array] * 5)
+        table.flush()
+
+    def write(self, sid, days, df):
+        """
+        Write the OHLCV data for the given sid.
+
+        If there is no bcolz ctable yet created for the sid, create it.
+
+        If the length of the bcolz ctable is not exactly to the date before
+        the first day provided, fill the ctable with 0s up to that date.
+
+        Writes in blocks of the size of the days times minutes per day.
+
+        Parameters:
+        -----------
+        sid : int
+            The asset identifer for the data being written.
+        days : pd.DatetimeIndex
```

Comment:
This was one of my questions in the PR notes, we could read this from the DataFrame, and probably should to reduce redundancy of data.

Also, we could require the DataFrame to have a `MultiIndex` of `(sid, dt)` and then walk through the DataFrame over sids using `get_level_values('sid')`.

Do you think using the `MultiIndex` would be too much of a hoop for users of the API to jump through?